sdk/include/opentelemetry/sdk/resource/resource.h

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include "opentelemetry/sdk/common/attribute_utils.h"
+#include "opentelemetry/sdk/resource/resource_detector.h"
+#include "opentelemetry/sdk/version/version.h"
+#include "opentelemetry/version.h"
+
+#include <memory>
+#include <sstream>
+#include <unordered_map>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace resource
+{
+
+using ResourceAttributes =
+    std::unordered_map<std::string, opentelemetry::sdk::common::OwnedAttributeValue>;
+
+class Resource
+{
+public:
+  const ResourceAttributes &GetAttributes() const noexcept;
+
+  /**
+   * Returns a new, merged {@link Resource} by merging the current Resource
+   * with the other Resource. In case of a collision, current Resource takes
+   * precedence.
+   *
+   * @param other the Resource that will be merged with this.
+   * @returns the newly merged Resource.
+   */
+
+  Resource Merge(const Resource &other) noexcept;
+
+  /**
+   * Returns a newly created Resource with the specified attributes.
+   * It adds (merge) SDK attributes and OTEL attributes before returning.
+   * @param attributes for this resource
+   * @returns the newly created Resource.
+   */
+
+  static Resource Create(const ResourceAttributes &attributes);
+
+  /**
+   * Returns an Empty resource.
+   */
+
+  static Resource &GetEmpty();
+
+  /**
+   * Returns a Resource that indentifies the SDK in use.
+   */
+
+  static Resource &GetDefault();
+
+protected:
```

Comment:
Is `protected` necessary or just private is fine?

Reply:
`protected` helps creating unit tests case - to subclass `Resource` and create instance of it for testing. I have modified the comment to indicate it's protected :)